tests/tests_pytorch/utilities/test_combined_loader.py

```diff
@@ -578,7 +588,129 @@ def test_combined_loader_can_be_pickled():
+        ("max_size", [[1, 2], [1, 2, 3]], 2, 2, True),
+        ("max_size", [[1, 2], [1, 2, 3]], 100, 3, True),  # limit exceeds largest iterable
+        ("max_size_cycle", [[1, 2], [1, 2, 3]], None, 2, False),
+        ("max_size_cycle", [[1, 2], [1, 2, 3]], 2, 2, True),
+        ("max_size_cycle", [[1, 2], [1, 2, 3]], 100, 3, True),  # limit exceeds largest iterable
+        ("sequential", [[1, 2], [1, 2, 3]], None, 2, False),
+        ("sequential", [[1, 2], [1, 2, 3]], 2, 2, False),
+        ("sequential", [[1, 2], [1, 2, 3]], 2, 4, True),  # limit in all iterables needs to be reached
+        ("sequential", [[1, 2], [1, 2, 3]], 100, 5, True),  # limit exceeds largest iterable
+        # unsized
+        ("min_size", [IterDataset()], None, 2, False),
+        ("min_size", [IterDataset()], None, 3, False),  # not sized, no prefetching -> can't know if done
+        ("min_size", [IterDataset()], 1, 1, True),
+        ("min_size", [IterDataset(2), IterDataset(3)], None, 1, False),
+        ("min_size", [IterDataset(2), IterDataset(3)], None, 2, False),  # not sized, no prefetching -> can't know
+        ("min_size", [IterDataset(2), IterDataset(3)], 1, 1, True),
+        ("max_size", [IterDataset(2), IterDataset(3)], None, 2, False),
+        ("max_size", [IterDataset(2), IterDataset(3)], 2, 2, True),
+        ("max_size", [IterDataset(2), IterDataset(3)], 100, 3, False),  # not sized, no prefetching -> can't know
+        ("max_size_cycle", [IterDataset(2), IterDataset(3)], None, 2, False),
+        ("max_size_cycle", [IterDataset(2), IterDataset(3)], 2, 2, True),
+        ("max_size_cycle", [IterDataset(2), IterDataset(3)], 100, 3, False),  # not sized, no prefetching -> can't know
+        ("sequential", [IterDataset(2), IterDataset(3)], None, 2, False),
+        ("sequential", [IterDataset(2), IterDataset(3)], 2, 2, False),  # not sized, no prefetching -> can't know
+        ("sequential", [IterDataset(2), IterDataset(3)], 2, 4, True),  # limit in all iterables needs to be reached
+        ("sequential", [IterDataset(2), IterDataset(3)], 100, 5, False),  # not sized, no prefetching -> can't know
+        # sized and unsized mixed
+        ("min_size", [[1, 2], IterDataset(3)], None, 1, False),
+        ("min_size", [[1, 2], IterDataset(3)], None, 2, True),  # smallest is sized -> done follows the limit
+        ("max_size", [IterDataset(2), [1, 2, 3]], None, 2, False),
+        ("max_size", [IterDataset(2), [1, 2, 3]], None, 3, False),  # 1st iterable is unsized -> can't know max
+        ("max_size_cycle", [IterDataset(2), [1, 2, 3]], None, 2, False),
+        ("max_size_cycle", [IterDataset(2), [1, 2, 3]], None, 3, False),
+        ("sequential", [[1, 2], IterDataset(3)], 2, 2, False),
+        ("sequential", [[1, 2], IterDataset(3)], 2, 4, True),  # limit in all iterables needs to be reached
+    ],
+)
+def test_done_dataloader_iter_with_limit(mode, iterables, limit, num_fetches, expected):
```

Comment:
There are similar tests to these ones in `tests/tests_pytorch/loops/test_fetchers.py`. I would suggest moving these together with the existing ones. Even though they use the combined_loader, they mostly assert the fetcher state